yqlplus_engine/src/main/java/com/yahoo/yqlplus/engine/internal/plan/types/base/MethodAssignableValue.java

```diff
@@ -23,14 +24,18 @@
     private final String name;
     private final int op;
     private final String desc;
-
-    public MethodAssignableValue(Method method, TypeWidget type, BytecodeExpression target) {
+    private TypeWidget castedReturntype;
+    public MethodAssignableValue(Method method, TypeWidget type, BytecodeExpression target, EngineValueTypeAdapter adapter) {
         this.type = type;
         this.target = target;
         this.ownerName = Type.getInternalName(method.getDeclaringClass());
         this.name = method.getName();
         this.op = method.getDeclaringClass().isInterface() ? Opcodes.INVOKEINTERFACE : Opcodes.INVOKEVIRTUAL;
         this.desc = Type.getMethodDescriptor(method);
+        TypeWidget returntype = adapter.adaptInternal(method.getReturnType());
```

Comment:
It may make sense to have this dance be a method on TypeAdapter -- I imagine there may be other places in the engine that miss this scenario (like the other property read cases, like Fields?).

Reply:
let me check.

Reply:
I combine the check and cast in CodeEmitter.